### ISSUE TITLE
dotnet watch and Blazor WebAssembly: client makes no attempt to connect to Hot Reload websocket server

### ISSUE THREAD

**Issue by mattbruv** [dotnet/aspnetcore#47340]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Describe the bug

`dotnet watch` hot reloading seems to randomly decide if it wants to work or not depending on which way the wind is blowing. Yesterday it was fine, today it is not working at all.

I'm running `dotnet watch --project . -v` in my project's client directory. On the surface it looks fine:

```
dotnet watch ⌚ Configuring the app to use browser-refresh middleware.
dotnet watch ⌚ Refresh server running at wss://localhost:61988,ws://localhost:61989.
dotnet watch 🔥 HotReloadProfile: BlazorWebAssembly.
```

When I edit a `.razor` file, my browser does not update and I get this output in the terminal only once:
```
dotnet watch ⌚ File changed: ./Pages/Users.razor.
```
Then the terminal does nothing until I kill it. Edits after this are not recognized or printed. Only when I press `^C` to end the process does it print something:

```
dotnet watch 🛑 Shutdown requested. Press Ctrl+C again to force exit.
dotnet watch ⌚ Failed to read Hot Reload capabilities: A task was canceled.
  ❔ Do you want to restart your app - Yes (y) / No (n) / Always (a) / Never (v)?
  ❔ dotnet watch ⌚ Caught top-level exception from hot reload: System.Threading.Tasks.TaskCanceledException: A task was canceled.
   at Microsoft.Extensions.Tools.Internal.ConsoleRequester.GetKeyAsync(String prompt, Func`2 validateInput, CancellationToken cancellationToken)
   at Microsoft.DotNet.Watcher.Tools.RudeEditDialog.EvaluateAsync(CancellationToken cancellationToken)
   at Microsoft.DotNet.Watcher.HotReloadDotNetWatcher.WatchAsync(DotNetWatchContext context, CancellationToken cancellationToken)
```

When I open my localhost web app and look at the `Network` tab in Chrome developer tools and filter for WebSocket connections, there are none. The client is not even attempting to connect to the hot reload server, nor are there any errors in my console to go off of. It just seems like it doesn't even try, and I have no idea how I'm supposed to remedy this.

### Expected Behavior

The client should make an attempt to connect to the WebSocket Refresh server when `dotnet watch` is run.
The client should communicate with the refresh server and correctly hot reload code on file changes.

### .NET Version

7.0.201

### Anything else?

```
.NET SDK:
 Version:   7.0.201
 Commit:    68f2d7e7a3

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  12.3
 OS Platform: Darwin
 RID:         osx.12-arm64
 Base Path:   /usr/local/share/dotnet/sdk/7.0.201/

Host:
  Version:      7.0.3
  Architecture: arm64
  Commit:       0a2bda10e8

.NET SDKs installed:
  6.0.406 [/usr/local/share/dotnet/sdk]
  7.0.201 [/usr/local/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.14 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.3 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.14 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.3 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]

Other architectures found:
  x64   [/usr/local/share/dotnet/x64]
    registered at [/etc/dotnet/install_location_x64]

Environment variables:
  Not set

global.json file:
  Not found
```

**Comment by paleocomburo** [comment]
I have the same problem. Found this issue by Googling the error message.

**Comment by paleocomburo** [comment]
Deleting all bin/ and obj/ folders inside the solution seems to have fixed it for now...

**Comment by hereisalex** [comment]
Also having this same problem, also found this issue by Googling the error message. What's the deal?

**Comment by amirvenus** [comment]
Anybody found a solution?

I get this on the output when running dotnet watch:

Failed to read Hot Reload capabilities: A task was cancelled

**Comment by tmat** [comment]
@mattbruv @amirvenus Please try the latest .NET 9 preview and file a new issue if you still see problems. We have made significant investments in reliability of dotnet-watch in this release.
